packages/cli/src/commands/bundle/assetCatalogIOS.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import path from 'path';
+import fs from 'fs-extra';
+import assetPathUtils from './assetPathUtils';
+import {AssetData} from './buildBundle';
+
+export async function cleanAssetCatalog(catalogDir: string): Promise<void> {
+  await Promise.all(
+    (await fs.readdir(catalogDir))
+      .filter((file) => file.endsWith('.imageset'))
+      .map((file) => fs.remove(file)),
+  );
+}
+
+type ImageSet = {
+  basePath: string;
+  files: {name: string; src: string; scale: number}[];
+};
+
+export function getImageSet(
+  catalogDir: string,
+  asset: AssetData,
+  scales: readonly number[],
+): ImageSet {
+  const fileName = assetPathUtils.getResourceIdentifier(asset);
+  return {
+    basePath: path.join(catalogDir, `${fileName}.imageset`),
+    files: scales.map((scale, idx) => {
+      const suffix = scale === 1 ? '' : `@${scale}x`;
+      return {
+        name: `${fileName + suffix}.${asset.type}`,
+        scale,
+        src: asset.files[idx],
+      };
+    }),
+  };
+}
+
+export function isCatalogAsset(asset: AssetData): boolean {
+  return asset.type === 'png' || asset.type === 'jpg' || asset.type === 'jpeg';
+}
+
+export async function writeImageSet(imageSet: ImageSet): Promise<void> {
+  await fs.mkdirs(imageSet.basePath);
+  await Promise.all(
+    imageSet.files.map(async (file) => {
+      const dest = path.join(imageSet.basePath, file.name);
+      await fs.copyFile(file.src, dest);
```

Comment:
we have a `copyFiles` utility, maybe worth reuse it?

Reply:
copyFiles copies all files in a directory so I don't think it could work here where we just want to copy specific files.